composition-js/src/merging/merge.ts

```diff
@@ -656,6 +658,10 @@ class Merger {
     // we want to make sure everything is ready.
     this.addMissingInterfaceObjectFieldsToImplementations();
 
+    // After converting some `@link`ed definitions to use `@join__directive`,
+    // we might have some imported scalars and input types to remove from the schema.
+    this.removeTypesAfterJoinDirectiveSerialization(this.merged);
```

Comment:
I'm a bit confused why we're merging subgraph types/directives that we later throw away here. (It sounds like we should update `isMergedType()` and `isMergedDirective()` to avoid merging them entirely.)

Reply:
Ah, this is the other action item I forgot about. @lennyburdette who should tackle looking into this?